MicrosoftCommonPropsHasBeenImported is set after importing Directory.Build.props

### Steps to reproduce

Importing `microsoft.build.centralpackageversions` SDK (2.0.52) from the Directory.Build.props.

Directory.Build.props
```xml
<Project>
  <Import Project="Sdk.props" Sdk="Microsoft.Build.CentralPackageVersions" />
</Project>
```

### Expected behavior

No warning is issued.

### Actual behavior

```
C:\Users\Andrew\.nuget\packages\microsoft.build.centralpackageversions\2.0.52\Sdk\Sdk.props(14,3): warning MSB4011: "C:\Program Files\dotnet\sdk\5.0.100-preview.5.20253.6\Current\Microsoft.Common.props" cannot be imported again. It was already imported at "C:\Program Files\dotnet\sdk\5.0.100-preview.5.20253.6\Sdks\Microsoft.NET.Sdk\Sdk\Sdk.props (40,3)". This is most likely a build authoring error. This subsequent import will be ignored. [D:\dev\SharpGenTools\SdkTests\Functions\Functions.csproj]
```

### Reason

`microsoft.build.centralpackageversions` correctly checks for `MicrosoftCommonPropsHasBeenImported`, but it's not set yet. `Microsoft.Common.props` sets it after importing `Directory.Build.props` file.

### Environment data

<details>
<summary>dotnet --info</summary>

```
.NET SDK (reflecting any global.json):
 Version:   5.0.100-preview.5.20253.6
 Commit:    7e7a355253

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19613
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.100-preview.5.20253.6\

Host (useful for support):
  Version: 5.0.0-preview.5.20253.6
  Commit:  6d40a8ff51

.NET SDKs installed:
  2.1.800 [C:\Program Files\dotnet\sdk]
  2.1.805 [C:\Program Files\dotnet\sdk]
  3.1.201 [C:\Program Files\dotnet\sdk]
  3.1.300-preview-015135 [C:\Program Files\dotnet\sdk]
  3.1.300-preview-015139 [C:\Program Files\dotnet\sdk]
  5.0.100-preview.3.20170.4 [C:\Program Files\dotnet\sdk]
  5.0.100-preview.5.20253.6 [C:\Program Files\dotnet\sdk]
```

</details>

<details>
<summary>MSBuild Binary Log</summary>

```
Process = "C:\Program Files\dotnet\dotnet.exe"
MSBuild executable path = "C:\Program Files\dotnet\sdk\5.0.100-preview.5.20253.6\MSBuild.dll"
Command line arguments = ""C:\Program Files\dotnet\sdk\5.0.100-preview.5.20253.6\MSBuild.dll" -maxcpucount -verbosity:m /t:GenerateSharpGenBindings /bl "-distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,C:\Program Files\dotnet\sdk\5.0.100-preview.5.20253.6\dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,C:\Program Files\dotnet\sdk\5.0.100-preview.5.20253.6\dotnet.dll""
Current directory = "D:\dev\SharpGenTools\SdkTests\Functions"
MSBuild version = "16.7.0-preview-20229-03+2cee6d020"
```

</details>

--- a/src/CentralPackageVersions.UnitTests/CentralPackageVersionsTests.cs
+++ b/src/CentralPackageVersions.UnitTests/CentralPackageVersionsTests.cs
@@ -236,6 +236,37 @@ namespace Microsoft.Build.CentralPackageVersions.UnitTests
             enableCentralPackageVersions.ShouldBe("false");
         }
 
+        [Fact]
+        public void LogErrorIfImportedInDirectoryBuildProps()
+        {
+            ProjectCreator.Create()
+                .Import(Path.Combine(ThisAssemblyDirectory, @"Sdk\Sdk.props"))
+                .Import(Path.Combine(ThisAssemblyDirectory, @"Sdk\Sdk.targets"))
+                .Save(GetTempFile("Directory.Build.props"));
+
+            ProjectCreator.Create()
+                .Save(GetTempFile("Directory.Build.targets"));
+
+            ProjectCreator.Templates
+                .PackagesProps(
+                    path: GetTempFile("Packages.props"),
+                    packageReferences: new Dictionary<string, string>
+                    {
+                        ["Foo"] = "1.2.3",
+                    })
+                .Save();
+
+            ProjectCreator.Templates
+                .SdkCsproj(projectCreator: creator => creator
+                    .ItemPackageReference("Foo"))
+                .Save(GetTempFile("Test.csproj"))
+                .TryBuild("CheckPackageReferences", out bool result, out BuildOutput buildOutput);
+
+            result.ShouldBeFalse(buildOutput.GetConsoleLog());
+
+            buildOutput.Errors.ShouldBe(new[] { "Microsoft.Build.CentralPackageVersions was not imported in Directory.Build.targets.  See https://github.com/microsoft/MSBuildSdks/tree/main/src/CentralPackageVersions for more information on how to include this SDK." }, buildOutput.GetConsoleLog());
+        }
+
         [Theory]
         [InlineData(".csproj")]
         [InlineData(".fsproj")]

--- a/src/CentralPackageVersions/Sdk/Sdk.props
+++ b/src/CentralPackageVersions/Sdk/Sdk.props
@@ -12,7 +12,5 @@
     <UsingMicrosoftCentralPackageVersionsSdk>true</UsingMicrosoftCentralPackageVersionsSdk>
   </PropertyGroup>
 
-  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition=" '$(MicrosoftCommonPropsHasBeenImported)' != 'true' And Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props') "/>
-
   <Import Project="$(CustomAfterCentralPackageVersionsProps)" Condition=" '$(CustomAfterCentralPackageVersionsProps)' != '' And Exists('$(CustomAfterCentralPackageVersionsProps)') " />
 </Project>

--- a/src/CentralPackageVersions/Sdk/Sdk.targets
+++ b/src/CentralPackageVersions/Sdk/Sdk.targets
@@ -14,6 +14,10 @@
 
   <PropertyGroup Condition=" '$(EnableCentralPackageVersions)' != 'false' ">
     <!--
+      Keep track if this file was imported before Directory.Build.targets
+    -->
+    <_WasMicrosoftCentralPackageVersionsSdkImportedInDirectoryBuildTargets Condition="'$(_WasMicrosoftCentralPackageVersionsSdkImportedInDirectoryBuildTargets)' == '' And '$(DirectoryBuildTargetsPath)' == ''">false</_WasMicrosoftCentralPackageVersionsSdkImportedInDirectoryBuildTargets>
+    <!--
       Walk up the directory tree looking for a Packages.props, unless a user has already specified a path.
     -->
     <CentralPackagesFile Condition=" '$(CentralPackagesFile)' == '' ">$([MSBuild]::GetPathOfFileAbove('Packages.props', $(MSBuildProjectDirectory)))</CentralPackagesFile>
@@ -151,7 +155,13 @@
       <_DuplicateGlobalPackageReference Include="@(_OriginalPackageReference)"
                                         Condition=" '@(GlobalPackageReference)' == '@(_OriginalPackageReference)' and '%(Identity)' != '' " />
     </ItemGroup>
-
+    
+    <!--
+      Log an error if this was imported before Directory.Build.targets
+    -->
+    <Error Text="Microsoft.Build.CentralPackageVersions was not imported in Directory.Build.targets.  See https://github.com/microsoft/MSBuildSdks/tree/main/src/CentralPackageVersions for more information on how to include this SDK."
+           Condition="'$(_WasMicrosoftCentralPackageVersionsSdkImportedInDirectoryBuildTargets)' == 'false'"
+           File="$(MSBuildProjectFullPath)" />
     <!--
       Log an error if there are any duplicate <PackageReference /> items where a <GlobalPackageReference /> is already defined.
     -->

